verifact/forum/schema.py

```diff
@@ -118,7 +126,53 @@ def mutate_and_get_payload(
 
         return AnswerCreate(answer=answer)
 
+class VoteUpdate(ClientIDMutation):
+    vote = Field(VoteNode)
+
+    class Input:
+        credible = Boolean() # if null, remove vote!
+        answer_id = ID()
+
+    @login_required
+    def mutate_and_get_payload(
+        self,
+        info,
+        answer_id,
+        **kwargs
+    ):
+        credible = kwargs.get('credible',None)
+        vote=None
+        viewer = info.context.user
+        try: # control flow could have been simpler -> if exists, delete, then create. but chose to do it this way for atomicity.
+            intansid = Node.get_node_from_global_id(
+                info,answer_id,only_type=AnswerNode
+            ).id
+            vote = Vote.objects.get(user=viewer,answer=intansid) # throws ObjectDoesNotExist if none found
+            if credible is None: # if vote null, delete
+                vote.delete()
+            else:
+                vote.credible = credible
+                vote.save()
+        except ObjectDoesNotExist as dne:
+            try:
+                if credible is not None:
+                    vote = Vote.objects.create(
+                        user=viewer,
+                        answer=Node.get_node_from_global_id(
+                            info, answer_id, only_type=AnswerNode
+                        ),
+                        credible=credible
+                    )
+            except AssertionError as ae:
+                if str(ae).startswith("Must receive an AnswerNode id."):
+                    raise GraphQLError(error_strings.ANSWER_ID_INVALID)
+                else:
```

Comment:
Don't need an `else` statement here, but I'm wondering if the whole thing can be removed.